docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,491 @@
+.. _build_for_ROCM_dpcpp:
+
+Building for ROCm
+~~~~~~~~~~~~~~~~~
+
+The ROCm backends can be enabled with ``ENABLE_ROCBLAS_BACKEND``, ``ENABLE_ROCFFT_BACKEND``, ``ENABLE_ROCSOLVER_BACKEND`` and ``ENABLE_ROCRAND_BACKEND``.
+
+For *RocBLAS*, *RocSOLVER* and *RocRAND*, the target device architecture must be set. This can be set with using the ``HIP_TARGETS`` parameter.
+For example, to enable a build for MI200 series GPUs, ``-DHIP_TARGETS=gfx90a`` should be set.
+Currently, DPC++ only supports a single HIP target. This may change in future versions.
+
+A few often-used architectures are listed below:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Architecture
+     - AMD GPU name
+   * - gfx90a
+     - AMD Instinct(TM) MI210/250/250X Accelerator
+   * - gfx908
+     - AMD Instinct(TM) MI 100 Accelerator
+   * - gfx906
+     - | AMD Radeon Instinct(TM) MI50/60 Accelerator
+       | AMD Radeon(TM) (Pro) VII Graphics Card
+   * - gfx900
+     - | Radeon Instinct(TM) MI 25 Accelerator
+       | Radeon(TM) RX Vega 64/56 Graphics
+
+For a host with ROCm installed, the device architecture can be retrieved via the ``rocminfo`` tool.
+The architecture will be displayed in the ``Name:`` row.
+
+.. _build_for_portblas_dpcpp:
+
+Building for portBLAS
+^^^^^^^^^^^^^^^^^^^^^^
+
+Note the portBLAS backend is experimental and currently only supports a
+subset of the operations and features. The portBLAS backend cannot be enabled
+with other backends and can only be used with the compile time dispatch.
+The portBLAS backend uses the `portBLAS <https://github.com/codeplaysoftware/portBLAS>`_
+project as a header-only library.
+
+* On Linux*
+
+.. code-block:: bash
+
+   # Inside <path to onemkl>
+   mkdir build && cd build
+   cmake .. -DENABLE_PORTBLAS_BACKEND=ON \
+            -DENABLE_MKLCPU_BACKEND=OFF  \
+            -DENABLE_MKLGPU_BACKEND=OFF  \
+            [-DREF_BLAS_ROOT=<reference_blas_install_prefix>] \ # required only for testing
+            [-DPORTBLAS_DIR=<path to portBLAS install directory>]
+   cmake --build .
+   ctest
+
+
+portBLAS will be downloaded automatically if not found.
+By default, the portBLAS backend is not tuned for any specific device.
+This tuning is required to achieve best performance.
+portBLAS can be tuned for a specific hardware target by adding compiler
+definitions in 2 ways:
+
+#.
+  Manually specify a tuning target with ``-DPORTBLAS_TUNING_TARGET=<target>``.
+  The list of portBLAS targets can be found
+  `here <https://github.com/codeplaysoftware/portBLAS#cmake-options>`_.
+  This will automatically set ``-fsycl-targets`` if needed.
+#.
+  If one target is set via ``-fsycl-targets`` the configuration step will
+  try to automatically detect the portBLAS tuning target. One can manually
+  specify ``-fsycl-targets`` via ``CMAKE_CXX_FLAGS``. See
+  `DPC++ User Manual <https://intel.github.io/llvm-docs/UsersManual.html>`_
+  for more information on ``-fsycl-targets``.
+
+.. _build_for_portfft_dpcpp:
+
+Building for portFFT
```

Comment:
Can we combine the portBLAS and portFFT sections? In my opinion they are the same in the same way that rocBLAS and rocFFT are the same.

Reply:
I've done this in [e2c341a](https://github.com/oneapi-src/oneMKL/pull/510/commits/e2c341a09be4d83c37db59bcba25d0e9b91a3c3e).